src/HassClient/Model/Selectors.cs

```diff
@@ -0,0 +1,96 @@
+﻿using JoySoftware.HomeAssistant.Helpers.Json;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
+
+namespace JoySoftware.HomeAssistant.Model
+{
+    public record ActionSelector
+    {
+    }
+
+    public record AddonSelector
+    {
+    }
+
+    public record AreaSelector
+    {
+        public DeviceSelector? Device { get; init; }
+
+        public EntitySelector? Entity { get; init; }
+    }
+
+    public record BooleanSelector
+    {
+    }
+
+    public record DeviceSelector
+    {
+        public string? Integration { get; init; }
+
+        public string? Manufacturer { get; init; }
+
+        public string? Model { get; init; }
+
+        public EntitySelector? Entity { get; init; }
+    }
+
+    public record EntitySelector
+    {
+        public string? Integration { get; init; }
+
+        public string? Domain { get; init; }
+
+        public string? DeviceClass { get; init; }
+    }
+
+    public record NumberSelector
+    {
+        [Required]
+        public double Min { get; init; }
+
+        [Required]
+        public double Max { get; init; }
+
+        public float? Step { get; init; }
+
+        public string? UnitOfMeasurement { get; init; }
+
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public NumberSelectorMode? Mode { get; init; }
```

Comment:
Could this break if HA adds a new mode later on, or will it have null?

Reply:
it will break, unfortunately. 
looking for a fix